src/components/profiles-modals/RenameProfileModal.vue

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import sanitize from 'sanitize-filename';
+import R2Error from "../../model/errors/R2Error";
+
+@Component({
+    components: {ModalCard}
+})
+export default class RenameProfileModal extends Vue {
+    private newProfileName: string = '';
+
+    get isOpen(): boolean {
+        this.newProfileName = this.$store.getters['profile/activeProfileName'];
+        return this.$store.state.modals.isRenameProfileModalOpen;
+    }
+
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+
+    closeModal() {
+        this.$store.commit('closeRenameProfileModal');
+    }
+
+    makeProfileNameSafe(nameToSanitize: string): string {
+        return sanitize(nameToSanitize);
+    }
+
+    doesProfileExist(nameToCheck: string): boolean {
+        if ((nameToCheck.match(new RegExp('^([a-zA-Z0-9])(\\s|[a-zA-Z0-9]|_|-|[.])*$'))) === null) {
+            return true;
+        }
+        const safe: string | undefined = sanitize(nameToCheck);
+        if (safe === undefined) {
+            return true;
+        }
+        return (this.profileList.find(
```

Comment:
Judgement call: Using `.some()` instead of `find()` would clean this up a bit. Currently the indentation of the last closing parenthesis makes this a bit wonky.